packages/core/src/Extension.ts

```diff
@@ -455,17 +456,13 @@ export class Extension<Options = any, Storage = any> {
   configure(options: Partial<Options> = {}) {
     // return a new instance so we can use the same extension
     // with different calls of `configure`
-    const extension = this.extend()
+    const extension = this.extend({
+      addOptions() {
+        return mergeDeep(this.parent?.() || {}, options) as Options
+      },
+    })
```

Comment:
This is the secret sauce, a configure is just a short way of expressing additional options to be applied